scripts/src/java/org/oppia/android/scripts/coverage/RunCoverage.kt

```diff
@@ -102,30 +106,100 @@ class RunCoverage(
    * @return a list of lists containing coverage data for each requested test target, if
    *     the file is exempted from having a test file, an empty list is returned
    */
-  fun execute() {
+  fun execute() = runBlocking {
     val testFileExemptionList = loadTestFileExemptionsProto(testFileExemptionTextProto)
       .testFileExemptionList
       .filter { it.testFileNotRequired }
       .map { it.exemptedFilePath }
 
     if (filePath in testFileExemptionList) {
       println("This file is exempted from having a test file; skipping coverage check.")
-      return
-    }
+    } else {
+      val testFilePaths = findTestFile(repoRoot, filePath)
+      if (testFilePaths.isEmpty()) {
+        error("No appropriate test file found for $filePath")
+      }
 
-    val testFilePaths = findTestFile(repoRoot, filePath)
-    if (testFilePaths.isEmpty()) {
-      error("No appropriate test file found for $filePath")
-    }
+      val testTargets = bazelClient.retrieveBazelTargets(testFilePaths)
 
-    val testTargets = bazelClient.retrieveBazelTargets(testFilePaths)
+      /*since I couldn't actually find any multi test target : file ones to test
+    * I am for now introducing mock data to test multi aggregated coverage report
+    * also that's going to save me a light year :|
+    * */
```

Comment:
@Rd4dev, could youbplease clarify what you are looking for? Is it a file with 2 seperate test files?

Reply:
Sure! In short it was not finding file with 2 separate test files but finding/having files with 2 separate test files that have coverages working on them. 

So, there are cases in app modules tests where one source file has 2 different test files (eg: **HomeActivity.kt** would be related to **HomeActivityTest.kt** (sharedTest) and **HomeActivityLocalTest.kt** (Test)), while computing coverages we will need to take account of both the test files and provide an aggregated coverage representation. 

With the [Migration to Bazel 6.5.0 PR](https://github.com/oppia/oppia-android/pull/4886#issuecomment-2135684503), while testing it is found that the coverages didn't work quite as expected for the app module files (few others too).

This was still a WIP. And I am not sure if any of the tests that fall in this n : 1 relationship work well with bazel coverage. But it is expected to be fixed and for now to get this PR work on some sample data I used the below provided proto to replicate the same behaviour.

So, to actually test with real source code files, I will either need to find if any of the corresponding files in this category work for coverages or look forward for the fix.